autoload/codefmt/clangformat.vim

```diff
@@ -120,32 +120,49 @@ function! codefmt#clangformat#GetFormatter() abort
       let l:cmd += ['-lines', l:startline . ':' . l:endline]
     endfor
 
+    let l:lines = getline(1, line('$'))
+
     " Version 3.4 introduced support for cursor tracking
     " http://llvm.org/releases/3.4/tools/clang/docs/ClangFormat.html
     let l:supports_cursor = s:ClangFormatHasAtLeastVersion([3, 4])
     if l:supports_cursor
-      " line2byte counts bytes from 1, and col counts from 1, so -2 
-      let l:cursor_pos = line2byte(line('.')) + col('.') - 2
+      " Avoid line2byte: https://github.com/vim/vim/issues/5930
+      let l:cursor_pos = col('.') - 1
+      for l:i in range(1, line('.') - 1)
+        let l:cursor_pos += len(l:lines[l:i - 1]) + 1
+      endfor
```

Comment:
Consider refactoring this into an `s:Func`. Later it might make a nice helper in itself in `maktaba#buffer#`.

Reply:
Sure thing! I'm not sure how useful these are apart from as a workaround for this bug, I'll let you make the call on moving them.